# Assignment 1 - Feature Engineering 

In [42]:
import pandas as pd
import numpy as np

## 1. Indetify and Rectify the data quality issues

In [43]:
## Reading the first data set using pandas
data_1 = pd.read_csv('bollywood.csv')
print(data_1.shape)
data_1.head()

(51, 11)


No      RelDate           MovieName  ReleaseTime         Genre Budget  \
0   1  Apr-18-2014            2 States            3  RomanceGenre   36Cr   
1   2   Jan-4-2013        Table No. 21            4     Thriller    Cr10   
2   3  Jul-18-2014  Amit Sahni Ki List            4   ComedyGenre   10Cr   
3   4   Jan-4-2013    Rajdhani Express            4    DramaGenre    7Cr   
4   5   Jul-4-2014        Bobby Jasoos           44   ComedyGenre   18Cr   

  BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  \
0                 104       8576361       26622.0           2527.0   
1                  12       1087320        1129.0            137.0   
2                   4        572336         586.0             54.0   
3                 383         42626          86.0             19.0   
4                10.8       3113427        4512.0           1224.0   

   YoutubeLikeDislikes  
0                29149  
1                 1266  
2                  640  
3                  105  
4                 5736

#### In the above results, we can see that there are 99 rows and 11 column

In [44]:
## Reading the second data set using pandas
data_2 = pd.read_csv('bollywood - 2.csv')
print(data_2.shape)
data_2.head()

(99, 10)


Movie_ID Release_Date             Movie_Name Release_Time     Genre  \
0        51     4-Jul-14  Lekar Hum Deewana Dil            N   Romance   
1        52    22-Mar-13                  Aatma            N  Thriller   
2        53     4-Apr-14         Main Tera Hero           LW    Comedy   
3        54    22-Mar-13               Sona Spa            N  Thriller   
4        55    29-Mar-13             Himmatwala            N    Action   

   BudgetCr  BoxOfficeCollectionCr  Youtube_Views  Youtube_Likes  \
0        16                   2.50        3998331           2307   
1         8                   8.78         841891            997   
2        40                  55.00        4903941          12657   
3        10                   0.24          79430             61   
4        50                 100.00        2501277              1   

   Youtube_Dislikes  
0               497  
1               306  
2              2139  
3               112  
4                 1

#### In the above results, we can see that there are 99 rows and 10 columns

#### Now when we have seen the 2 data sets, we have seen few errors with data and we will try to correct one by one

##### 1.1 In the first data set, there is an extra column 'YoutubeLikeDislikes' which is the sum of 2 columns 'YoutubeLikes' and 'YoutubeDislikes'
##### we can remove this extra column or add the column is dataset 2 to bring it to the same level with the second data set
##### we will go wuth the approach of creating this column and adding in the data set 2

In [45]:
data_1.drop(['YoutubeLikeDislikes'], axis = 1, inplace = True) 
print(data_1.shape)
data_1.head()

(51, 10)


No      RelDate           MovieName  ReleaseTime         Genre Budget  \
0   1  Apr-18-2014            2 States            3  RomanceGenre   36Cr   
1   2   Jan-4-2013        Table No. 21            4     Thriller    Cr10   
2   3  Jul-18-2014  Amit Sahni Ki List            4   ComedyGenre   10Cr   
3   4   Jan-4-2013    Rajdhani Express            4    DramaGenre    7Cr   
4   5   Jul-4-2014        Bobby Jasoos           44   ComedyGenre   18Cr   

  BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
0                 104       8576361       26622.0           2527.0  
1                  12       1087320        1129.0            137.0  
2                   4        572336         586.0             54.0  
3                 383         42626          86.0             19.0  
4                10.8       3113427        4512.0           1224.0

##### 1.2. Next error we see is in the column names, both the data sets have different column name although the data represented as same, so we can correct the column names and apply the same column names to both the data set. Once this is done, we can club the two data sets and make it one 

In [46]:
corrected_col_names = ['MovieId','ReleaseDate','MovieName','ReleaseTime','Genre','Budget',
                       'BoxOfficeCollection','YoutubeViews','YoutubeLikes','YoutubeDislikes']
print(corrected_col_names)

['MovieId', 'ReleaseDate', 'MovieName', 'ReleaseTime', 'Genre', 'Budget', 'BoxOfficeCollection', 'YoutubeViews', 'YoutubeLikes', 'YoutubeDislikes']


In [47]:
data_1.columns = corrected_col_names
data_2.columns = corrected_col_names

In [48]:
data = data_1.append(data_2)
data.reset_index(inplace=True, drop=True)
print(data.shape)
data.head(200)

(150, 10)


MovieId  ReleaseDate                    MovieName ReleaseTime  \
0          1  Apr-18-2014                     2 States           3   
1          2   Jan-4-2013                 Table No. 21           4   
2          3  Jul-18-2014           Amit Sahni Ki List           4   
3          4   Jan-4-2013             Rajdhani Express           4   
4          5   Jul-4-2014                 Bobby Jasoos          44   
..       ...          ...                          ...         ...   
145      145    27-Feb-15           Dum Laga Ke Haisha           N   
146      146    13-Mar-15                         NH10           N   
147      147    20-Mar-15  Dilliwali Zaalim Girlfriend           N   
148      148    20-Mar-15                     Hunterrr           N   
149      149    23-May-14                Kochadaiiyaan          HS   

            Genre Budget BoxOfficeCollection  YoutubeViews  YoutubeLikes  \
0    RomanceGenre   36Cr                 104       8576361       26622.0   
1       Thriller    Cr10                  12       1087320        1129.0   
2     ComedyGenre   10Cr                   4        572336         586.0   
3      DramaGenre    7Cr                 383         42626          86.0   
4     ComedyGenre   18Cr                10.8       3113427        4512.0   
..            ...    ...                 ...           ...           ...   
145        Comedy     15                  30       3250917        8185.0   
146      Thriller     13                32.1       5592977       15464.0   
147        Comedy     32                  12       2316047        4289.0   
148        Comedy      5               11.89       4674795        3706.0   
149        Action    150                 120       4740727       13466.0   

     YoutubeDislikes  
0             2527.0  
1              137.0  
2               54.0  
3               19.0  
4             1224.0  
..               ...  
145            615.0  
146           1513.0  
147            807.0  
148            762.0  
149           2649.0  

[150 rows x 10 columns]

###### Observation: after merging the two data sets now we have 150 rows and 10 columns

##### let's check some basic attributes now

In [49]:
data.isnull().sum()

MovieId                0
ReleaseDate            0
MovieName              0
ReleaseTime            0
Genre                  0
Budget                 0
BoxOfficeCollection    0
YoutubeViews           0
YoutubeLikes           1
YoutubeDislikes        3
dtype: int64

In [50]:
data.isnull().mean()

MovieId                0.000000
ReleaseDate            0.000000
MovieName              0.000000
ReleaseTime            0.000000
Genre                  0.000000
Budget                 0.000000
BoxOfficeCollection    0.000000
YoutubeViews           0.000000
YoutubeLikes           0.006667
YoutubeDislikes        0.020000
dtype: float64

#### 1.3. We see that there two columns which are Null (NaN). we have to solve this issue.
#####         a.  the column 'YoutubeLikes' has total 1 null value i.e. )0.66% of total rows 
#####         b. the column 'YoutubeDislikes' has total 3 null values i.e. 2% of total rows.. 

In [51]:
## function to replace variable (x) with value (y)
def replaceNaN(df, x, y):
    df[x]=df[x].fillna(y)       

In [52]:
# Filling missing values with medians of the columns
median_ytlikes = data['YoutubeLikes'].median()
print(' Mean of YoutubeLikes =>', median_ytlikes)
replaceNaN(data, 'YoutubeLikes', median_ytlikes)

# Filling missing values with medians of the columns
median_ytdislikes = data['YoutubeDislikes'].median()
print(' Mean of YoutubeDislikes =>', median_ytdislikes)
replaceNaN(data, 'YoutubeDislikes', median_ytdislikes)

 Mean of YoutubeLikes => 4111.0
 Mean of YoutubeDislikes => 614.0


In [53]:
# after imputation, there should not be nay missing values
data.isnull().sum()

MovieId                0
ReleaseDate            0
MovieName              0
ReleaseTime            0
Genre                  0
Budget                 0
BoxOfficeCollection    0
YoutubeViews           0
YoutubeLikes           0
YoutubeDislikes        0
dtype: int64

##### we can see above that now we have zero such coulmns, where the value is missing

##### 1.4 In the dataset we observe that the column 'Budget' has 'Cr' has come as prefix or suffix. So we can remove the literal 'Cr', 'CR', 'cr'

In [54]:
## function to replace value (x) with value (y)
def replaceVal(data, x, y):
    data.replace(to_replace =x, value = y, regex = True, inplace=True)

In [55]:
replaceVal(data['Budget'], 'Cr', '')
data['Budget']=data.Budget.astype(float)

replaceVal(data['BoxOfficeCollection'], 'Cr', '')
data['BoxOfficeCollection']=data.BoxOfficeCollection.astype(float)

##### 1.5 The column 'Genre' has string literal 'Genre' in some rows, so we need to remove it

In [56]:
replaceVal(data['Genre'], 'Genre', '')

## to remove leading or trailing spaces from Genre
data['Genre'] = data['Genre'].apply(lambda _: str(_).strip())
data['Genre']=data.Genre.astype(str)

# replaceVal(data['Genre'], 'Romance', '1')
# replaceVal(data['Genre'], 'Drama', '2')
# replaceVal(data['Genre'], 'Action', '3')
# replaceVal(data['Genre'], 'Comedy', '4')
# replaceVal(data['Genre'], 'Thriller', '4')

##### 1.6 The column 'ReleaseDate' has some invalid dates, we need to correct that column

In [57]:
from dateutil.parser import parse

def checkInvalidDate(movie_id, dt):
    try:
        dt = dt.strip()
        return parse(dt).date()
    except ValueError as e:
        print(f"Invalid date '{dt}' for the movie id '{movie_id}' - reason: {e}")
        return dt

In [58]:
def validateDate(data, dt):
    dt = dt.strip()
    return parse(dt)

In [59]:
data.apply(lambda x: checkInvalidDate(x.MovieId, str(x.ReleaseDate)), axis=1)

Invalid date 'Sqp-19-2014' for the movie id '7' - reason: Unknown string format: Sqp-19-2014
Invalid date 'Jum-27-2014' for the movie id '13' - reason: Unknown string format: Jum-27-2014
Invalid date 'juu-6-2014' for the movie id '18' - reason: Unknown string format: juu-6-2014


0      2014-04-18
1      2013-01-04
2      2014-07-18
3      2013-01-04
4      2014-07-04
          ...    
145    2015-02-27
146    2015-03-13
147    2015-03-20
148    2015-03-20
149    2014-05-23
Length: 150, dtype: object

###### As we see the results, we have three invalid dates which we need to correct

In [60]:
replaceVal(data['ReleaseDate'], 'Sqp', 'Sep')
replaceVal(data['ReleaseDate'], 'Jum', 'Jun')
replaceVal(data['ReleaseDate'], 'juu', 'Jun')

# we can confrm that the dates are now valid
data.apply(lambda x: checkInvalidDate(x.MovieId, str(x.ReleaseDate)), axis=1)

0      2014-04-18
1      2013-01-04
2      2014-07-18
3      2013-01-04
4      2014-07-04
          ...    
145    2015-02-27
146    2015-03-13
147    2015-03-20
148    2015-03-20
149    2014-05-23
Length: 150, dtype: object

##### 1.7 The column 'ReleaseTime' has some invalid values. We need to correct that column

In [61]:
def checkInvalidString(movie_id, val):
    try:
        if isinstance(int(val), int):
            return val
        else:
            print(f"Invalid String value '{val}' for the movie id '{movie_id}'")
    except ValueError as e :
        print(f"Invalid String value '{val}' for the movie id '{movie_id}'")
        return val

In [62]:
def checkInvalidNumber(movie_id, val):
    try:
        if isinstance(int(val), int):
            return val
        else:
            print(f"Invalid Int value '{val}' for the movie id '{movie_id}'")
    except ValueError as e :
        print(f"Invalid Int value '{val}' for the movie id '{movie_id}'")
        return val

In [63]:
data.apply(lambda x: checkInvalidString(x.MovieId, str(x.ReleaseTime)), axis=1)

Invalid String value 'N' for the movie id '51'
Invalid String value 'N' for the movie id '52'
Invalid String value 'LW' for the movie id '53'
Invalid String value 'N' for the movie id '54'
Invalid String value 'N' for the movie id '55'
Invalid String value 'HS' for the movie id '56'
Invalid String value 'N' for the movie id '57'
Invalid String value 'N' for the movie id '58'
Invalid String value 'N' for the movie id '59'
Invalid String value 'LW' for the movie id '60'
Invalid String value 'HS' for the movie id '61'
Invalid String value 'LW' for the movie id '62'
Invalid String value 'FS' for the movie id '63'
Invalid String value 'N' for the movie id '64'
Invalid String value 'N' for the movie id '65'
Invalid String value 'N' for the movie id '66'
Invalid String value 'N' for the movie id '67'
Invalid String value 'HS' for the movie id '68'
Invalid String value 'N' for the movie id '69'
Invalid String value 'N' for the movie id '70'
Invalid String value 'N' for the movie id '71'
Invali

0       3
1       4
2       4
3       4
4      44
       ..
145     N
146     N
147     N
148     N
149    HS
Length: 150, dtype: object

##### we can see above that we have so many rows with faulty 'ReleaseTime', such as 'HS', 'LW', 'N', 'FS'.
##### after cleaning, we should not see any invalid value

In [64]:
replaceVal(data['ReleaseTime'], 'LW', '1')
replaceVal(data['ReleaseTime'], 'HS', '2')
replaceVal(data['ReleaseTime'], 'N', '3')
replaceVal(data['ReleaseTime'], 'FS', '4')

# we can confrm that the dates are now valid
data.apply(lambda x: checkInvalidString(x.MovieId, str(x.ReleaseTime)), axis=1)

0       3
1       4
2       4
3       4
4      44
       ..
145     3
146     3
147     3
148     3
149     2
Length: 150, dtype: object

##### 1.8 Removing duplicates

In [65]:
data[data.duplicated(['MovieId'])]

MovieId ReleaseDate              MovieName ReleaseTime    Genre  Budget  \
51       51    4-Jul-14  Lekar Hum Deewana Dil           3  Romance    16.0   

    BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
51                  2.5       3998331        2307.0            497.0

In [66]:
# to find out is there any duplicate by column Movie_Name
data[data.duplicated(['MovieName'])]

MovieId ReleaseDate              MovieName ReleaseTime    Genre  Budget  \
51       51    4-Jul-14  Lekar Hum Deewana Dil           3  Romance    16.0   

    BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
51                  2.5       3998331        2307.0            497.0

In [67]:
# from the above 2 statements, movie_id = 51 are duplicated
data[data.MovieId == 51]

MovieId ReleaseDate              MovieName ReleaseTime    Genre  Budget  \
50       51  Jul-7-2014  Lekar Hum Deewana Dil           4  Romance    16.0   
51       51    4-Jul-14  Lekar Hum Deewana Dil           3  Romance    16.0   

    BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
50                  2.5       3998331        2307.0            497.0  
51                  2.5       3998331        2307.0            497.0

In [68]:
# but the movie_id = 51 has two different release_date
# also YoutubeLikeDislikes is updated in index = 51
# so lets drop row with index = 50 and update release_date with imdb scrapper
data.drop_duplicates(subset ="MovieId", keep = 'first', inplace = True)

In [69]:
data[data.MovieId == 51]

MovieId ReleaseDate              MovieName ReleaseTime    Genre  Budget  \
50       51  Jul-7-2014  Lekar Hum Deewana Dil           4  Romance    16.0   

    BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
50                  2.5       3998331        2307.0            497.0

In [70]:
print(data.shape)
data.head()

(149, 10)


MovieId  ReleaseDate           MovieName ReleaseTime     Genre  Budget  \
0        1  Apr-18-2014            2 States           3   Romance    36.0   
1        2   Jan-4-2013        Table No. 21           4  Thriller    10.0   
2        3  Jul-18-2014  Amit Sahni Ki List           4    Comedy    10.0   
3        4   Jan-4-2013    Rajdhani Express           4     Drama     7.0   
4        5   Jul-4-2014        Bobby Jasoos          44    Comedy    18.0   

   BoxOfficeCollection  YoutubeViews  YoutubeLikes  YoutubeDislikes  
0                104.0       8576361       26622.0           2527.0  
1                 12.0       1087320        1129.0            137.0  
2                  4.0        572336         586.0             54.0  
3                383.0         42626          86.0             19.0  
4                 10.8       3113427        4512.0           1224.0

#### After all the data cleaning steps above, data looks cleaner than earlier

## 2. To find records present in the dataset and metadata information of dataset

In [71]:
## given data set 1
print(data_1.shape)
data_1.describe()

(51, 10)


MovieId  ReleaseTime  YoutubeViews   YoutubeLikes  YoutubeDislikes
count  51.000000    51.000000  5.100000e+01      50.000000        48.000000
mean   26.000000     4.137255  3.504813e+06    8779.680000      1326.104167
std    14.866069     5.779341  3.547764e+06   15449.317168      2133.718842
min     1.000000     1.000000  4.354000e+03       1.000000         1.000000
25%    13.500000     2.500000  1.059242e+06    1558.500000       170.750000
50%    26.000000     4.000000  2.375050e+06    3817.000000       562.000000
75%    38.500000     4.000000  5.008115e+06   12063.250000      1657.250000
max    51.000000    44.000000  1.898622e+07  101275.000000     10269.000000

In [72]:
## given data set 2
print(data_2.shape)
data_2.describe()

(99, 10)


MovieId      Budget  BoxOfficeCollection  YoutubeViews  \
count   99.000000   99.000000            99.000000  9.900000e+01   
mean   100.000000   30.080808            56.253939  3.258616e+06   
std     28.722813   28.485828           101.429241  3.479723e+06   
min     51.000000    2.000000             0.240000  5.142000e+03   
25%     75.500000   12.000000             8.390000  1.140044e+06   
50%    100.000000   22.000000            29.000000  2.501277e+06   
75%    124.500000   35.000000            55.000000  4.185784e+06   
max    149.000000  150.000000           735.000000  2.317107e+07   

       YoutubeLikes  YoutubeDislikes  
count     99.000000        99.000000  
mean    7445.202020      1170.989899  
std    11187.769162      1722.899736  
min        1.000000         1.000000  
25%     1099.000000       261.500000  
50%     4111.000000       628.000000  
75%     8589.000000      1378.000000  
max    72070.000000     11888.000000

In [73]:
# after merging the 2 datasets
print(data.shape)
data.info()

(149, 10)
<class 'pandas.core.frame.DataFrame'>
Int64Index: 149 entries, 0 to 149
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   MovieId              149 non-null    int64  
 1   ReleaseDate          149 non-null    object 
 2   MovieName            149 non-null    object 
 3   ReleaseTime          149 non-null    object 
 4   Genre                149 non-null    object 
 5   Budget               149 non-null    float64
 6   BoxOfficeCollection  149 non-null    float64
 7   YoutubeViews         149 non-null    int64  
 8   YoutubeLikes         149 non-null    float64
 9   YoutubeDislikes      149 non-null    float64
dtypes: float64(4), int64(2), object(4)
memory usage: 10.5+ KB


## 3. How many movies in each genre got released in different release times?

In [74]:
## Find the Unique Genre values
data['Genre'].unique

<bound method Series.unique of 0       Romance
1      Thriller
2        Comedy
3         Drama
4        Comedy
         ...   
145      Comedy
146    Thriller
147      Comedy
148      Comedy
149      Action
Name: Genre, Length: 149, dtype: object>

In [75]:
# Group the Genre values based on ReleaseTime
data.groupby(['Genre','ReleaseTime']).sum()

MovieId  Budget  BoxOfficeCollection  YoutubeViews  \
Genre    ReleaseTime                                                       
Action   2                 86    58.0               131.00       7433473   
         4                208   369.0               602.89      32259701   
         1                272   187.0               239.20      13989483   
         2                149   150.0               120.00       4740727   
         3                710   325.0               390.25      34577280   
         4                357   278.0               717.00      43610929   
Comedy   2                 78    95.0                82.30       9613073   
         3                  9    31.0                27.00       2333067   
         4                147   163.0               334.20      21704004   
         44                 5    18.0                10.80       3113427   
         1                343   169.0               409.50      13228028   
         2                147     7.0                 5.00       4815001   
         3               1542   334.0               387.34      34560447   
         4                369    93.0               404.00       1945982   
Drama    1                 65    39.0                91.00       8224700   
         2                 86    25.0                45.50       2333660   
         3                 27   125.0               383.00       6395202   
         4                137   123.0               619.46      10850254   
         2                280   100.0               758.75      14218298   
         3               1745   379.0               702.70      43890198   
         4                257    56.0                85.00       7613220   
Romance  2                 46    56.0               170.50       8524443   
         3                  1    36.0               104.00       8576361   
         4                215   136.0               239.65      24702386   
         1                297    59.0                97.13       3966730   
         2                 68    10.0                 1.10       1537063   
         3                855   258.0               634.65      27318788   
         4                373    74.0                74.00      12256792   
Thriller 1                 32    12.0                25.00       2249055   
         3                 17    60.0               162.00       7641333   
         4                167    79.0                82.13      22791315   
         1                115    29.0                53.00       3049275   
         2                100    16.0                29.00       3707889   
         3               1568   299.0               332.92      44553596   
         4                302   139.0               126.10       5024887   

                      YoutubeLikes  YoutubeDislikes  
Genre    ReleaseTime                                 
Action   2                 14529.0           2028.0  
         4                130921.0          14756.0  
         1                 33164.0           4383.0  
         2                 13466.0           2649.0  
         3                 65316.0           8927.0  
         4                144477.0          19064.0  
Comedy   2                  8682.0            730.0  
         3                  2436.0            591.0  
         4                 50962.0           8458.0  
         44                 4512.0           1224.0  
         1                 27367.0          11367.0  
         2                  4685.0           2287.0  
         3                 69720.0          10087.0  
         4                  7996.0           1604.0  
Drama    1                 24627.0           2632.0  
         2                  5201.0            242.0  
         3                 37585.0          10269.0  
         4                 31730.0           3472.0  
         2                 46398.0           5884.0  
         3                107404.0          10594.0  
         4           

## 4. Which month of the year, maximum movie releases are seen?

In [76]:
from dateutil.parser import parse

data['ReleaseDate'] = data['ReleaseDate'].apply(lambda x: pd.to_datetime(parse(x).date(), errors='coerce'))

In [77]:
## Now finding the count of movies release in Monthly frequency
max_release_year = data.set_index('ReleaseDate').groupby(pd.Grouper(freq="M")).agg('count')['MovieId']

print('Maximum movies released =>', str(max_release_year.max()))
print(str.format('Maximum movies released in the Month of {0} and Year {1}',str(max_release_year.idxmax().date().strftime("%B")),str(max_release_year.idxmax().date().year)))

Maximum movies released => 12
Maximum movies released in the Month of May and Year 2014


## 5. Which month of the year typically sees most releases of high budgeted movies, that is, movies with budget of 25 crore or more?

In [78]:
## Now finding the count of movies release in Monthly frequency

max_release_year = data[data['Budget'] > 25].set_index('ReleaseDate').groupby(pd.Grouper(freq="M")).agg('count')['MovieId']

print('Maximum movies released =>', str(max_release_year.max()))
print(str.format('Maximum movies released in the Month of {0} and Year {1}',str(max_release_year.idxmax().date().strftime("%B")),str(max_release_year.idxmax().date().year)))

Maximum movies released => 5
Maximum movies released in the Month of July and Year 2013


## 6. Which are the top 10 movies with maximum return of investment (ROI)? Calculate ROI as (Box office collection – Budget) / Budget

In [79]:
# calculate ROI
roi = (data['BoxOfficeCollection'] - data['Budget']) / data['Budget']

# Now we have ROI as a colum, we can use it to find top 10 movies by max ROI
data['ROI'] = roi

data.sort_values(['ROI'], ascending=False)[0:10][['MovieId','MovieName','ROI']].iloc[0:, 1:]

MovieName        ROI
3             Rajdhani Express  53.714286
65                  Aashiqui 2   8.166667
90                          PK   7.647059
133                Grand Masti   7.514286
136               The Lunchbox   7.500000
88                      Fukrey   6.240000
59                    Mary Kom   5.933333
129                     Shahid   5.666667
37   Humpty Sharma Ki Dulhania   5.500000
102         Bhaag Milkha Bhaag   4.466667

## END